corehq/apps/domain/tests/test_auth.py

```diff
@@ -157,6 +157,13 @@ def test_cannot_use_expired_key(self):
         with travel(datetime(year=2020, month=3, day=11), tick=False):
             self.assertIsNone(self.backend.authenticate(request, 'test@dimagi.com', '1234'))
 
+    def test_domain_scoped_api_key_allows_authentication_to_domain_agnostic_urls(self):
+        user = self._create_user('test@dimagi.com')
+        self._create_api_key_for_user(user, key='1234', domain='test-domain')
```

Comment:
Just left a comment on this about the last commit, but are you certain this is acceptable?  My expectation when domain-scoping an API key is that it can't be used to access resources outside the scope of that domain.  Isn't this a violation of that?  Like, couldn't I use this to create a new API key that isn't domain scoped and then use that on other domains?

Reply:
I _thought_ this sounded familiar.  Looks like I had the same feedback on the PR that introduced that:
https://github.com/dimagi/commcare-hq/pull/35225#issuecomment-2422877591

I'll raise it in #gtd-security-discussion to see if I'm off-base here.